### PR TITLE
Patch to get InjectManifest to work

### PR DESCRIPTION
**Pull request by zoetrope69**
This is what I'm doing to get InjectManifest to work at the moment. 

Related to #1

### CODE REVIEW

File: inject-manifest.js [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
+const replaceDefaultPlugin = require('./replace-default-plugin');
 const {InjectManifest} = require("workbox-webpack-plugin");
 const getDefaultConfig = require('./default-config');
-const getESMDefaultConfig = require('./esm-config');
 
 module.exports = function(config, helpers, workboxConfig) {
   const defaultConfig = getDefaultConfig(config);
   delete defaultConfig['navigateFallback'];
   delete defaultConfig['navigateFallbackWhitelist'];
-  const esmConfig = getESMDefaultConfig(config);
```
Comment (zaccolley) [inline inject-manifest.js]: We never send this through

File: replace-default-plugin.js [hunk 2]
```diff
@@ -6,7 +6,10 @@ module.exports = function (config, helpers, newPlugin, esmPlugin) {
       config.plugins.splice(index, 1, newPlugin);
     }
   });
-  if (helpers.getPluginsByName(config, 'BabelEsmPlugin')) {
+  if (pluginInstances.length === 0) {
+    config.plugins.push(newPlugin);
+  }
+  if (esmPlugin && helpers.getPluginsByName(config, 'BabelEsmPlugin')) {
```
Comment (zaccolley) [inline replace-default-plugin.js]: This is the bit of code that errored, so checking if we send the esmPlugin through stops it from erroring

File: replace-default-plugin.js [hunk 3]
```diff
@@ -6,7 +6,10 @@ module.exports = function (config, helpers, newPlugin, esmPlugin) {
       config.plugins.splice(index, 1, newPlugin);
     }
   });
-  if (helpers.getPluginsByName(config, 'BabelEsmPlugin')) {
+  if (pluginInstances.length === 0) {
```
Comment (zaccolley) [inline replace-default-plugin.js]: I had to add this for the service worker file to get generated on the other end, looks like `SWPrecacheWebpackPlugin` is missing so it never replaces it

File: inject-manifest.js (hunk 1 above)
Comment (prateekbh) [reply inject-manifest.js]: we should, shouldn't we?

File: replace-default-plugin.js (hunk 3 above)
Comment (prateekbh) [reply replace-default-plugin.js]: wait! what cli version are you using it?

**Pull request closed without merging**